easybuild/easyblocks/f/freefem.py

```diff
@@ -0,0 +1,165 @@
+
+        configopts = [
+            '--disable-optim',  # disable custom optimizations (not needed, $CFLAGS set by EasyBuild is picked up)
+            '--enable-download',  # enable downloading of dependencies
+            '--disable-openblas',  # do not download OpenBLAS
+        ]
+
+        blas_family = self.toolchain.blas_family()
+        if blas_family == toolchain.OPENBLAS:
+            configopts.append("--with-blas=openblas")
+        elif blas_family == toolchain.INTELMKL:
+            mkl_root = get_software_root('imkl')
+            configopts.append("--with-mkl=%s" % os.path.join(mkl_root, 'mkl', 'lib', 'intel64'))
+
+        # check depencies and add the corresponding configure options for FreeFEM
+        hdf5_root = get_software_root('HDF5')
+        if hdf5_root:
+            configopts.append('--with-hdf5=%s ' % os.path.join(hdf5_root, 'bin', 'h5pcc'))
+
+        gsl_root = get_software_root('GSL')
+        if gsl_root:
+            configopts.append('--with-gsl-prefix=%s' % gsl_root)
+
+        petsc_root = get_software_root('PETSc')
+        if petsc_root:
+            configopts.append('--with-petsc=%s' % os.path.join(petsc_root, 'lib', 'petsc', 'conf', 'petscvariables'))
+
+        bemtool_root = get_software_root('BemTool')
+        if bemtool_root:
+            configopts.append('--with-bem-include=%s' % bemtool_root)
+
+        for configopt in configopts:
+            self.cfg.update('configopts', configopt)
+
+        # initial configuration
+        out = super(EB_FreeFEM, self).configure_step()
+
+        regex = re.compile("WARNING: unrecognized options: (.*)", re.M)
+        res = regex.search(out)
+        if res:
+            raise EasyBuildError("One or more configure options not recognized: %s" % res.group(1))
+
+        if not petsc_root:
+            # re-create installation dir (deletes old installation),
+            # then set keeppreviousinstall to True (to avoid deleting PETSc installation)
+            self.make_installdir()
+            self.cfg['keeppreviousinstall'] = True
+
+            # configure and make petsc-slepc
+            # download & build PETSc as recommended by FreeFEM if no PETSc dependency was provided
+            cwd = change_dir(os.path.join('3rdparty', 'ff-petsc'))
+
+            cmd = ['make']
+            if self.cfg['parallel']:
+                cmd.append('-j %s' % self.cfg['parallel'])
+            cmd.append('petsc-slepc')
+
+            run_cmd(' '.join(cmd), log_all=True, simple=False)
+
+            change_dir(cwd)
+
+            # reconfigure for FreeFEM build
+            super(EB_FreeFEM, self).configure_step()
+
+    def test_step(self):
+        """Run tests."""
+
+        # run tests unless they're disabled explicitly
+        if self.cfg['runtest'] is None or self.cfg['runtest']:
+            # avoid oversubscribing, by using both OpenMP threads and having OpenBLAS use threads as well
+            env.setvar('OPENBLAS_NUM_THREAD', '1')
```

Comment:
@boegel I think this should be OPENBLAS_NUM_THREADS ? (added 'S')

Reply:
Yes, needs the 'S'

Reply:
And it should be OMP_NUM_THREADS since we build OpenBLAS the way we do.